lib/expression/callbacks/standard.ex

```diff
@@ -100,6 +114,74 @@ defmodule Expression.Callbacks.Standard do
     end
   end
 
+  @doc """
+  Calculates a new datetime based on a base date, a desired day and a desired time.
+
+  If no base date is supplied, it will fallback to current time.
+  """
+  @expression_doc doc:
+                    "Shifts a datetime to the next occurrence of a given day of the week and a set time.",
+                  expression: "next(\"monday\", \"10:00\", datetime)",
+                  context: %{
+                    "datetime" => ~U[2023-02-03 20:18:03Z],
+                    "number" => %{"from_addr" => "+5521987724322"}
+                  },
+                  fake_result: Map.merge(@datetime_struct, %{day: 6, hour: 10, minute: 0})
+  @expression_doc doc:
+                    "If the day of the week is the same as the base date, the next occurrence is 7 days later.",
+                  expression: "next(\"thursday\", \"10:30\", datetime)",
+                  context: %{
+                    "datetime" => ~U[2023-02-02 20:18:03Z],
+                    "number" => %{"from_addr" => "+5521987724322"}
+                  },
+                  fake_result: Map.merge(@datetime_struct, %{day: 9, hour: 10, minute: 30})
+  @spec next(map, {:literal, String.t()}, {:literal, String.t()}, any) :: DateTime.t()
+  def next(
+        %{"number" => %{"from_addr" => from_addr}} = ctx,
+        desired_day,
+        time,
+        base_date \\ DateTime.utc_now()
+      ) do
+    base_date =
+      if is_struct(base_date) == false,
+        do: extract_datetimeish(eval!(base_date, ctx)),
+        else: base_date
+
+    [desired_day, time] = eval_args!([desired_day, time], ctx)
+
+    [hour, minute] = time |> String.split(":") |> Enum.map(&String.to_integer/1)
```

Comment:
There are some parsers in [Expression.DateHelpers](https://github.com/turnhub/expression/blob/446f3a352fe3d3a4335aea4ef907d22bf4eb5e01/lib/date_helpers.ex#L79) that parse a datetime. Those are using an internal parser called `iso_time()` and `us_time()` -- maybe we can split those into into separate things and parse `10:30` into a %Time{} straight away?

Reply:
Happy to treat this as scope creep though! it's up to you. The current solution works fine as well.

Reply:
At first I did something like that, but for the Timex.set function I need the hour and minutes to be separated, so putting them into a time struct doesn't work.